Source/WebCore/svg/SVGPathSegImpl.h

```diff
@@ -222,3 +222,23 @@ class SVGPathSegCurvetoCubicSmoothRel final : public SVGPathSegCurvetoCubicSmoot
 };
 
 } // namespace WebCore
+
+SPECIALIZE_TYPE_TRAITS_SVGPATHSEG(SVGPathSegClosePath, ClosePath);
+SPECIALIZE_TYPE_TRAITS_SVGPATHSEG(SVGPathSegLinetoHorizontalAbs, LineToHorizontalAbs);
+SPECIALIZE_TYPE_TRAITS_SVGPATHSEG(SVGPathSegLinetoHorizontalRel, LineToHorizontalRel);
+SPECIALIZE_TYPE_TRAITS_SVGPATHSEG(SVGPathSegLinetoVerticalAbs, LineToVerticalAbs);
+SPECIALIZE_TYPE_TRAITS_SVGPATHSEG(SVGPathSegLinetoVerticalRel, LineToVerticalRel);
+SPECIALIZE_TYPE_TRAITS_SVGPATHSEG(SVGPathSegMovetoAbs, MoveToAbs);
+SPECIALIZE_TYPE_TRAITS_SVGPATHSEG(SVGPathSegMovetoRel, MoveToRel);
+SPECIALIZE_TYPE_TRAITS_SVGPATHSEG(SVGPathSegLinetoAbs, LineToAbs);
+SPECIALIZE_TYPE_TRAITS_SVGPATHSEG(SVGPathSegLinetoRel, LineToRel);
+SPECIALIZE_TYPE_TRAITS_SVGPATHSEG(SVGPathSegCurvetoQuadraticAbs, CurveToQuadraticAbs);
+SPECIALIZE_TYPE_TRAITS_SVGPATHSEG(SVGPathSegCurvetoQuadraticRel, CurveToQuadraticRel);
+SPECIALIZE_TYPE_TRAITS_SVGPATHSEG(SVGPathSegCurvetoCubicAbs, CurveToCubicAbs);
+SPECIALIZE_TYPE_TRAITS_SVGPATHSEG(SVGPathSegCurvetoCubicRel, CurveToCubicRel);
+SPECIALIZE_TYPE_TRAITS_SVGPATHSEG(SVGPathSegArcAbs, ArcAbs);
+SPECIALIZE_TYPE_TRAITS_SVGPATHSEG(SVGPathSegArcRel, ArcRel);
+SPECIALIZE_TYPE_TRAITS_SVGPATHSEG(SVGPathSegCurvetoQuadraticSmoothAbs, CurveToQuadraticSmoothAbs);
+SPECIALIZE_TYPE_TRAITS_SVGPATHSEG(SVGPathSegCurvetoQuadraticSmoothRel, CurveToQuadraticSmoothRel);
+SPECIALIZE_TYPE_TRAITS_SVGPATHSEG(SVGPathSegCurvetoCubicSmoothAbs, CurveToCubicSmoothAbs);
+SPECIALIZE_TYPE_TRAITS_SVGPATHSEG(SVGPathSegCurvetoCubicSmoothRel, CurveToCubicSmoothRel);
```

Comment:
Why do we need to add these traits? The removed code seems to work without it. I think createWrapper() uses `static_reference_cast<>`. Does the new binding generated code use `downcast<>`?

Reply:
The generated bindings code uses `is<>()` / `downcast<>()`, which is why we need those traits. It needs it because it gets a base class pointer and it needs to figure out which subclass it actually is, in order to construct the right JS wrapper.